includes/gateway/class-omise-payment-paynow.php

```diff
@@ -70,16 +83,19 @@ public function display_qrcode( $order, $context = 'view' ) {
 		$charge_id = $this->get_charge_id_from_order();
 		$charge    = OmiseCharge::retrieve( $charge_id );
 		$qrcode    = $charge['source']['scannable_code']['image']['download_uri'];
-		?>
-		<div class="omise omise-paynow-details" <?php echo 'email' === $context ? 'style="margin-bottom: 4em; text-align:center;"' : ''; ?>>
-			<div class="omise omise-paynow-logo"></div>
-			<p>
-				<?php echo __( 'Scan the QR code to pay', 'omise' ); ?>
-			</p>
-			<div class="omise omise-paynow-qrcode">
-				<img src="<?php echo $qrcode; ?>" alt="Omise QR code ID: <?php echo $charge['source']['scannable_code']['image']['id']; ?>">
+
+		if ( 'view' === $context ) : ?>
+			<div class="omise omise-paynow-details" <?php echo 'email' === $context ? 'style="margin-bottom: 4em; text-align:center;"' : ''; ?>>
+				<div class="omise omise-paynow-logo"></div>
+				<p>
+					<?php echo __( 'Scan the QR code to pay', 'omise' ); ?>
+				</p>
+				<div class="omise omise-paynow-qrcode">
+					<img src="<?php echo $qrcode; ?>" alt="Omise QR code ID: <?php echo $charge['source']['scannable_code']['image']['id']; ?>">
```

Comment:
This link downloads the QR image directly which is likely isn't standard flow. Should we create a public link to new page and display QR?

Reply:
I actually was 50/50 about this... One side kinda assume that to pay a QR code, probably buyer does not need to open an entire website to scan a code (also it's easier to implement). Another side, for its credibility, maybe we need to render the webpage.

What do you think?